hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/command/CommandStatusReportHandler.java

```diff
@@ -55,13 +53,21 @@ public void onMessage(CommandStatusReportFromDatanode report,
     cmdStatusList.forEach(cmdStatus -> {
       LOGGER.trace("Emitting command status for id:{} type: {}", cmdStatus
           .getCmdId(), cmdStatus.getType());
-      if (cmdStatus.getType() == SCMCommandProto.Type.deleteBlocksCommand) {
+      switch (cmdStatus.getType()) {
+      case deleteBlocksCommand:
         if (cmdStatus.getStatus() == CommandStatus.Status.EXECUTED) {
           publisher.fireEvent(SCMEvents.DELETE_BLOCK_STATUS,
               new DeleteBlockStatus(cmdStatus));
         }
-      } else {
-        LOGGER.debug("CommandStatus of type:{} not handled in " +
+        break;
+      case createPipelineCommand:
+        if (cmdStatus.getStatus() != CommandStatus.Status.PENDING) {
+          publisher.fireEvent(SCMEvents.CREATE_PIPELINE_STATUS,
+              new CreatePipelineStatus(cmdStatus));
+        }
```

Comment:
Do we need a handler for destoryPipeline?

Reply:
Same as below